test/spanningtrees/wilson.jl

```diff
@@ -0,0 +1,84 @@
+#test that the result of random_spanning_tree is correct
+#ie. it should be a spanning forest, oriented towards the roots 
+function check_correctness(edges,roots)
+    F = SimpleDiGraph(edges)
+    nmissing = max(0,Int.(maximum(roots))-nv(F)) #there might be some missing nodes if they're alone in their components
+    add_vertices!(F,nmissing)
+    @test !is_cyclic(F)
+    @test all(outdegree(F,roots) .== 0)
+    #test that all nodes lead to a root
+    for cc in connected_components(F)
+        if (length(cc)==1)
+            @test cc[1] in roots
+        else
+            rc = intersect(cc,roots)
+            #there should be a single root per connected component
+            @test length(rc)==1
+            rc = rc[1]
+            bf=bfs_parents(F,rc;dir=-1)[cc]
+            @test all(bf .> 0)
+        end
+    end
+end
+#rather than import from StatsBase
+function relcounts(values :: Array{Int,1},k :: Int)
+    cc=zeros(k)
+    for v in values
+        cc[v] += 1
+    end
+    cc ./ length(values)
+end
+
+@testset "Wilson" begin
+    gg = Grid([4,4])
+    for g in testgraphs(gg)
+        rt = @inferred(random_spanning_tree(g))
+        check_correctness(rt.edges,rt.roots)
+    end
+
+    #Try a bunch of random graphs
+    rg = [SimpleGraph(20,20) for i in 1:15]
```

Comment:
This is not a great test approach because it's not deterministic. Can you create some specific graphs instead?

Reply:
Fixed